clientAkkaHttp/src/main/scala/RequestBuilder.scala

```diff
@@ -42,13 +43,27 @@ class RequestBuilder(
   private[this] def splitTokenArgs(args: Map[String, Json]): (Map[String, Json], Map[String, Json]) =
     args.partition { case (_, value) => value.as[wiro.Auth].isRight }
 
+  private[this] def splitHeaderArgs(args: Map[String, Json]): (Map[String, Json], Map[String, Json]) =
+    args.partition { case (_, value) => value.as[wiro.OperationParameters].isRight }
+
   private[this] def handleAuth(tokenCandidates: List[Json]): List[RawHeader] =
     if (tokenCandidates.length > 1)
       throw new Exception("Only one parameter of wiro.Auth type should be provided")
     else tokenCandidates.map(_.as[wiro.Auth]).collect {
       case Right(wiro.Auth(token)) => RawHeader("Authorization", s"Token token=$token")
     }
 
+  private[this] def handleHeaders(headersCandidates: List[Json]): List[RawHeader] =
+    headersCandidates.headOption match {
+      case Some(operationParameters) =>
+        operationParameters.as[wiro.OperationParameters] match {
+          case Right(wiro.OperationParameters(parameters)) =>
+            parameters.map { case (headerName, headerValue) => RawHeader(headerName, headerValue.noSpaces) }.toList
+          case Left(_) => Nil
+        }
+      case None => Nil
+    }
```

Comment:
I find it somewhat difficult to read this nested pattern matching, but I suspect that the alternative using something like `Option.flatMap` and `Either.toOption` would be even less readable.